src/base/bittorrent/session.cpp

```diff
@@ -1149,7 +1150,17 @@ void Session::initializeNativeSession()
     loadLTSettings(pack);
     lt::session_params sessionParams {pack, {}};
 #ifdef QBT_USES_LIBTORRENT2
-    sessionParams.disk_io_constructor = customDiskIOConstructor;
+    switch (storageType())
```

Comment:
@glassez 
Can't we move this so it can be reconfigured on user's request?

Reply:
> @glassez Can't we move this so it can be reconfigured on user's request?

No. It is part of immutable session parameters that are accepted only in construct time.